bundle/edu.gemini.itc.shared/src/main/scala/edu/gemini/itc/shared/ItcService.scala

```diff
@@ -76,13 +87,29 @@ final case class ItcSpectroscopyResult(charts: List[SpcChartData], warnings: Lis
     * This method will fail if the result (chart/data) you're looking for does not exist.
     */
   def allSeries(ct: SpcChartType, dt: SpcDataType): List[SpcSeriesData] = chart(ct).allSeries(dt)
+
+  def peakPixelFlux(ccd: Int = 0): Int = {
+    // zip signal and background values, sum them and return max value (i.e. max(signal + background))
+    def maxSum(s: SpcSeriesData, b: SpcSeriesData) =
+      s.yValues.zip(b.yValues).map(p => p._1 + p._2).max.round
+
+    // zip signal and background value arrays and return max of all maximums
+    // e.g. GNIRS with cross dispersion will have several arrays for signal and background, one for each order
+    def maxAllSum(s: List[SpcSeriesData], b: List[SpcSeriesData]) =
+      s.zip(b).map(p => maxSum(p._1, p._2)).max
+
+    val signal     = allSeries(SignalChart, SignalData)
+    val background = allSeries(SignalChart, BackgroundData)
+    maxAllSum(signal, background).toInt
+  }
+
 }
 
 object ItcSpectroscopyResult {
 
   // java compatibility
-  def apply(charts: java.util.List[SpcChartData], warnings: java.util.List[ItcWarning]) =
-    new ItcSpectroscopyResult(charts.toList, warnings.toList)
+  def apply(charts: java.util.List[SpcChartData], warnings: List[ItcWarning]) =
```

Comment:
Right, the second parameter is created with Scala code, however the method is called from Java, so that's where the mix comes from. With ITC I am following very much a step-by-step clean-up and migrate to Scala where doable/useful strategy, which leads to quite a mix of Scala/Java code in some places. This will get better over time..